src/Microsoft.AspNetCore.Mvc.Core/Internal/ContentResultExecutor.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class ContentResultExecutor
+    {
+        /// <summary>
+        /// The maximum number of characters that are encoded and written to stream in a single pass.
+        /// </summary>
+        public const int MaxCharacterChunkSize = 1024;
+
+        private readonly string DefaultContentType = new MediaTypeHeaderValue("text/plain")
+        {
+            Encoding = Encoding.UTF8
+        }.ToString();
+
+        private readonly ILogger<ContentResultExecutor> _logger;
+
+        public ContentResultExecutor(ILogger<ContentResultExecutor> logger)
+        {
+            _logger = logger;
+        }
+
+        public async Task ExecuteAsync(ActionContext context, ContentResult result)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            if (result == null)
+            {
+                throw new ArgumentNullException(nameof(result));
+            }
+
+            var response = context.HttpContext.Response;
+
+            string resolvedContentType;
+            Encoding resolvedContentTypeEncoding;
+            ResponseContentTypeHelper.ResolveContentTypeAndEncoding(
```

Comment:
I had a chance to catch up with @lodejard yesterday, his suggestion is centered around changing the behavior to not have a default content type. 

Please open a separate issue for that discussion, that's a much larger topic than what we're doing in this PR.